### PR TITLE
committed task_01, task_02

### PR DESCRIPTION
**Pull request by elenaalkoli**

### CODE REVIEW

File: hw_11/task_02.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+type TObjects = { [key: string]: string | number | boolean }; //объект или массив объектов
+
+interface IValueTypes {
+  string: number;
+  number: number;
+  boolean: number;
+}
+
+function CountValueTypes(input: TObjects | TObjects[]): IValueTypes {
```
Comment (VladimirLQA) [inline hw_11/task_02.ts:91]: According to the book "Javascript: the good parts", you should only capitalise the first character of the name of a function when you need to construct the object by "new" keyword.

This is called "the Constructor Invocation Pattern", a way to inherits
```suggestion
function countValueTypes(input: TObjects | TObjects[]): IValueTypes {
```

File: hw_11/task_02.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  const result: IValueTypes = {    //объект куда будем складывать каунты
+    string: 0,
+    number: 0,
+    boolean: 0,
+  };
+
+  function countTypes(input: TObjects) {
```
Comment (VladimirLQA) [inline hw_11/task_02.ts:98]: You can move this function outside and use as helper, but not neccessary :)

**Review by VladimirLQA — APPROVED** [review]

File: hw_11/task_02.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    flat: number;
+}
+interface IEmployee {
+    name: string;
+    surname: string;
+    readonly salary: number;
+    address?: IAddress;
+}
+
+function isItEmployee(employee: ItEmployee | IEmployee): employee is ItEmployee {
```
Comment (anatoly-karpovich) [inline hw_11/task_02.ts:39]: awesome!

File: hw_11/task_02.ts (hunk 2 above)
Comment (anatoly-karpovich) [reply hw_11/task_02.ts:98]: Definetely it is necessary. Especially when function is connected to parent function's "state", i mean result variable. helper functions should not have any dependency on such variables and should return calculated data

**Review by anatoly-karpovich — APPROVED** [review]

**Comment by anatoly-karpovich** [comment]
well done!

**Pull request merged**